cores/esp8266/Stream.h

```diff
@@ -102,12 +113,104 @@ class Stream: public Print {
+
+        // (outputTimeoutPossible() is defined in Print::)
+
+        //////////////////// extensions: Streaming streams to streams
+        // Stream::to*()
+        //
+        // Stream::to*() uses 1-copy transfers when peekBuffer API is
+        // available, or makes a regular transfer through a temporary buffer.
+        //
+        // - for efficiency, Stream classes should implement peekAPI when
+        //   possible
+        // - for an efficient timeout management, Print/Stream classes
+        //   should implement {output,input}TimeoutPossible()
+
+        using oneShotMs = esp8266::polledTimeout::oneShotFastMs;
+
+        // ::to*() methods:
+        // - always stop before timeout when "no-more-input-possible-data"
+        //   or "no-more-output-possible-data" condition is met
+        // - always return number of transfered bytes
+        // When result is 0 or less than requested maxLen, this->getLastTo()
+        // contains an error reason.
+
+        // transfers already buffered / immediately available data (no timeout)
+        // returns number of transfered bytes
+        size_t toNow (Print* to) { return toFull(to, -1, -1, oneShotMs::alwaysExpired); }
+        size_t toNow (Print& to) { return toNow(&to); }
+
+        // transfers data until timeout
+        // returns number of transfered bytes
+        size_t toAll (Print* to, const oneShotMs::timeType timeoutMs = oneShotMs::neverExpires) { return toFull(to, -1, -1, timeoutMs); }
+        size_t toAll (Print& to, const oneShotMs::timeType timeoutMs = oneShotMs::neverExpires) { return toAll(&to, timeoutMs); }
+
+        // transfers data until a char is encountered (the char is swallowed but not transfered) with timeout
+        // returns number of transfered bytes
+        size_t toUntil (Print* to, const int readUntilChar, const oneShotMs::timeType timeoutMs = oneShotMs::neverExpires) { return toFull(to, -1, readUntilChar, timeoutMs); }
+        size_t toUntil (Print& to, const int readUntilChar, const oneShotMs::timeType timeoutMs = oneShotMs::neverExpires) { return toUntil(&to, readUntilChar, timeoutMs); }
+
+        // transfers data until requested size or timeout
+        // returns number of transfered bytes
+        size_t toSize (Print* to, const ssize_t maxLen, const oneShotMs::timeType timeoutMs = oneShotMs::neverExpires) { return toFull(to, maxLen, -1, timeoutMs); }
+        size_t toSize (Print& to, const ssize_t maxLen, const oneShotMs::timeType timeoutMs = oneShotMs::neverExpires) { return toSize(&to, maxLen, timeoutMs); }
+
+        // remaining size (-1 by default = unknown)
+        virtual ssize_t streamSize () { return -1; }
+
+        typedef enum
+        {
+            STREAMTO_SUCCESS = 0,
+            STREAMTO_TIMED_OUT,
+            STREAMTO_READ_ERROR,
+            STREAMTO_WRITE_ERROR,
+            STREAMTO_SHORT,
+        } toReport_e;
+
+        toReport_e getLastTo () /*const*/ { return (toReport_e)getWriteError(); }
```

Comment:
Think this needs the work "error" in the fcn name...i.e. `getLastSendError` (see below) or `getLastToError`.